hardware_interface/src/system.cpp

```diff
@@ -196,16 +196,26 @@ std::vector<CommandInterface> System::export_command_interfaces()
 
 return_type System::prepare_command_mode_switch(
   const std::vector<std::string> & start_interfaces,
-  const std::vector<std::string> & stop_interfaces)
+  const std::vector<std::string> & stop_interfaces,
+  const std::vector<std::string> & command_data)
 {
-  return impl_->prepare_command_mode_switch(start_interfaces, stop_interfaces);
+  if (command_data.empty())
```

Comment:
I think this case shouldn't be handled at this level. We don't check here for empty start_/stop_interfaces either. The check should be done within the actual HW interface implementation to keep the code here as simple as possible.

Reply:
This check is here to ensure backwards compatibility. All implementations of the hardware interface until this point use the 2 argument function signature. If we don't have this check and instead only use the 3 argument function signature, then this will break all implementations of the hardware interface that expect the 2 argument signature.

By checking to see if command_data is empty, that ensures that only those using this new functionality have to modify their hardware interface implementation to use the 3 argument signature.